### PR TITLE
Make typeof(null) be null, not null

### LINKED ISSUE #246: Make typeof(null) be "null", not null
See https://github.com/taskcluster/json-e/pull/245 where it turns out that this function has some unfortunate behaviors.

### PR DESCRIPTION
**Pull request by alexandrasp**
I change typeof functionality in builtins.js to return null as a string. Fix #246 

# Checklist

Before submitting a pull request, please check the following:

* [x] All tests pass for all implementations (all implementations must behave identically)
* [x] New tests have been added for any new functionality or to prevent regressions of a bugfix
* [x] Added a short description of the change to `newsfragments/$issue.bugfix` (for fixes) or `.feature` (for new features) or `.doc` (for documentation-only changes)

### CODE REVIEW

**Comment by djmitche** [comment]
Good start -- this will need changes in `specification.yml`, and there are three more languages to fix :)

File: jsone/render.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def interpolate(string, context):
             if isinstance(parsed, (list, dict)):
                 raise TemplateError(
                     "interpolation of '{}' produced an array or object".format(string[:offset]))
-            if to_str(parsed) == "null":
+            if to_str(parsed) == "null" and parsed != "null":
```
Comment (alexandrasp) [inline jsone/render.py]: really hard figure out that parse should be different of `null` here!

Comment (djmitche) [reply jsone/render.py]: There are two cases where `to_str(parsed)` is "null": `parsed=None` and `parsed="null"`.  So I think this statement is equivalent to `if parsed is None`.  To check, I made that change and ran the tests and they still passed.

**Comment by djmitche** [comment]
Released in v3.0.0

**Pull request merged**